torch/nn/utils/stateless.py

```diff
@@ -88,68 +94,89 @@ def _untie_named_tensors_map(
     return untied_parameters_and_buffers
 
 
-@contextlib.contextmanager
-def _reparametrize_module(
-    module: "torch.nn.Module",
-    parameters_and_buffers: Dict[str, Tensor],
-    *,
-    tie_weights: bool = False,
-    strict: bool = False,
-    stack_weights: bool = False,
-) -> Iterator[None]:
-    if tie_weights:
-        untied_parameters_and_buffers = _untie_named_tensors_map(
-            module, parameters_and_buffers
-        )
-    else:
-        untied_parameters_and_buffers = parameters_and_buffers
+class ReparametrizeModule:
```

Comment:
make this private, it's in a public file (torch.nn.utils.stateless) but it isn't public API